src/Lecture2.hs

```diff
@@ -70,7 +74,14 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt index nums
+      | index < 0 = (Nothing, nums)
+      | index >= length part1 = (Nothing, nums)
```

Comment:
Unfortunately, `length` hangs on infinite lists so this implementation of `removeAt` doesn't work on infinite lists ♾️